src/Microsoft.AspNetCore.SignalR.Client.Core/HubConnection.cs

```diff
@@ -111,6 +134,20 @@ private void ResetTimeoutTimer()
 
         private async Task StartAsyncCore()
         {
+            SafeAssert(_connectionLock.CurrentCount == 0, "We're not in the Connection Lock!");
+
+            CheckDisposed();
+
+            if (_connection != null)
+            {
+                // We have an existing connection!
+                throw new InvalidOperationException($"The '{nameof(StartAsync)}' method cannot be called if the connection has already been started.");
+            }
+
+            _connection = _connectionFactory();
+            _connection.OnReceived((data, state) => ((HubConnection)state).OnDataReceivedAsync(data), this);
+            _connection.Closed += (c, e) => _ = OnClosed(c, e);
```

Comment:
What is underscore here? Maybe place a comment with what is going on for the uninformed reader.

Reply:
Adding the comment now. We generally do this pattern instead of `async void` for fire-and-forget tasks.